test/unit/kubernetes-deploy/kubernetes_resource/deployment_test.rb

```diff
@@ -2,118 +2,332 @@
+      "readyReplicas" => 1
     }
+    replica_sets = [build_rs_template(status: rs_status)]
+
+    dep_template = build_deployment_template(status: deployment_status,
+      rollout: 'maxUnavailable', max_unavailable: '100%')
+    deploy = build_synced_deployment(template: dep_template, replica_sets: replica_sets)
+    assert deploy.deploy_succeeded?
+
+    # rounds up to two max
+    deploy = build_synced_deployment(
+      template: build_deployment_template(status: deployment_status, rollout: 'maxUnavailable', max_unavailable: '67%'),
+      replica_sets: replica_sets
+    )
+    assert deploy.deploy_succeeded?
 
-    deploy = KubernetesDeploy::Deployment.new(namespace: "", context: "", logger: logger, definition: rollout)
-    mock_rs = Minitest::Mock.new
-    needed = 2
-    mock_rs.expect :present?, true
-    mock_rs.expect :desired_replicas, needed
-    mock_rs.expect :ready_replicas, needed - 1
-    mock_rs.expect :available_replicas, needed - 1
-    deploy.instance_variable_set(:@max_unavailable, 0)
-    deploy.instance_variable_set(:@latest_rs, mock_rs)
-    deploy.instance_variable_set(:@desired_replicas, needed)
+    # rounds down to one max
+    deploy = build_synced_deployment(
+      template: build_deployment_template(status: deployment_status, rollout: 'maxUnavailable', max_unavailable: '66%'),
+      replica_sets: replica_sets
+    )
+    refute deploy.deploy_succeeded?
 
+    deploy = build_synced_deployment(
+      template: build_deployment_template(status: deployment_status, rollout: 'maxUnavailable', max_unavailable: '0%'),
+      replica_sets: replica_sets
+    )
     refute deploy.deploy_succeeded?
   end
 
-  def test_deploy_succeeded_fails_with_max_unavailable_as_a_percent
-    rollout = {
-      'metadata' => {
-        'name' => 'fake',
-        'annotations' => { KubernetesDeploy::Deployment::REQUIRED_ROLLOUT_ANNOTATION => 'maxUnavailable' }
-      }
-    }
+  def test_deploy_succeeded_raises_with_invalid_rollout_annotation
+    deploy = build_synced_deployment(
+      template: build_deployment_template(rollout: 'bad'),
+      replica_sets: [build_rs_template]
+    )
+    msg = "'#{KubernetesDeploy::Deployment::REQUIRED_ROLLOUT_ANNOTATION}: bad' is "\
+      "invalid. Acceptable values: maxUnavailable, full, none"
+    assert_raises_message(KubernetesDeploy::FatalDeploymentError, msg) do
+      deploy.deploy_succeeded?
+    end
+  end
+
+  def test_validation_fails_with_invalid_rollout_annotation
+    deploy = build_synced_deployment(template: build_deployment_template(rollout: 'bad'), replica_sets: [])
+    deploy.kubectl.expects(:run).with('create', '-f', anything, '--dry-run', '--output=name', anything).returns(
+      ["", "super failed", SystemExit.new(1)]
+    )
+    refute deploy.validate_definition
+
+    expected = <<~STRING.strip
+      super failed
+      '#{KubernetesDeploy::Deployment::REQUIRED_ROLLOUT_ANNOTATION}: bad' is invalid. Acceptable values: maxUnavailable, full, none
```

Comment:
Perhaps:

```
 Acceptable values: #{REQUIRED_ROLLOUT_TYPES.join(', ')}
```

... alternatively don't test for "Acceptable values" at all?

Reply:
Danny had the same suggestion. My opinion is that the assertion isn't as valuable if it is pretty much the same as the originating code. By writing it out the way the user would read it, you can notice things like spacing/punctuation errors much more easily. But I'm clearly in the  minority on this one so I'll change it. 😄

Reply:
> My opinion is that the assertion isn't as valuable if it is pretty much the same as the originating code. 

True, which is why I wonder if it's worthwhile checking for the "Acceptable values".
Effectively that tests that the list hasn't changed (and also punctuation).

I'm ok with it either way.